<launch>
  <!-- Arguments -->
  <arg name="use_gui" default="false" doc="Enable joint state publisher GUI"/>
  <arg name="use_rviz" default="true" doc="Start RViz"/>
  
  <!-- Load robot description -->
  <param name="robot_description" 
         textfile="$(find 3dof_rrr_robot_arm)/urdf/3DoF_RRR_Robot_Arm.urdf"/>
  
  <!-- Load SRDF -->
  <param name="robot_description_semantic" 
         textfile="$(find 3dof_rrr_robot_arm)/config/3DoF_RRR_Robot_Arm.srdf"/>
  
  <!-- Load kinematics config -->
  <rosparam command="load" file="$(find 3dof_rrr_robot_arm)/config/kinematics.yaml" 
            ns="robot_description_kinematics"/>
  
  <!-- Load joint limits -->
  <rosparam command="load" file="$(find 3dof_rrr_robot_arm)/config/joint_limits.yaml" 
            ns="robot_description_planning"/>
  
  <!-- Robot State Publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
  
  <!-- Joint State Publisher (with or without GUI) -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" unless="$(arg use_gui)">
    <param name="use_gui" value="false"/>
  </node>
  
  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" if="$(arg use_gui)"/>
  
  <!-- Fake Joint Trajectory Action Server -->
  <node name="fake_joint_trajectory_action_server" pkg="3dof_rrr_robot_arm" 
        type="fake_joint_trajectory_action_server.py" output="screen"/>
  
  <!-- MoveIt Move Group -->
  <include file="$(find 3dof_rrr_robot_arm)/launch/move_group.launch"/>
  
  <!-- RViz -->
  <node name="rviz" pkg="rviz" type="rviz" if="$(arg use_rviz)"
        args="-d $(find 3dof_rrr_robot_arm)/config/moveit.rviz" output="screen"/>
        
</launch>
